token-swap/program/src/curve/calculator.rs

```diff
@@ -219,7 +229,7 @@ pub mod test {
                 swap_token_b_amount,
                 pool_supply,
                 trade_direction,
-                RoundDirection::Floor,
+                LiquidityProviderOperation::Withdrawal,
```

Comment:
I'm now remembering why I kept this as `RoundDirection` and not the operation -- it's not always clear which one to use!  For example, if I'm depositing token A and B to get pool tokens, we actually specify the number of pool tokens and then calculate the amount of token A and B necessary to get that.  I call `pool_tokens_to_trading_tokens`, but since it's a deposit, I want to be sure to `ceiling` the amount of A and B that are taken.  When I deposit one side, I specify the amount of token A to deposit and then calculate the amount of pool tokens to receive, which means I want to `floor` that calculation to avoid minting too many pool tokens.  All that to say, these should all be `Deposit`.  

This will be fine for now, but it will be very confusing once we add more flexible options for deposits.  Or we'll just have to add more `LiquidityProviderOperation` options

Reply:
done